docs/zpure/index.md

```diff
@@ -0,0 +1,348 @@
+    x <- one
+    y <- two
+    z <- three
+  } yield x + y + z
+```
+
+The easiest way to run a computation is to use the `run` method. It requires us to have already provided any required environment and initial state and handled our errors and just produces the final value.
+
+```scala mdoc
+val value: Int =
+  three.run
+```
+
+## Working With Errors
+
+Since `ZIO` also has an error type the operators for working with errors are quite similar to the ones on `ZIO`.
+
+We can create a computation that fails with an error using the `fail` operator.
+
+```scala mdoc
+val fail: ZPure[Nothing, Unit, Unit, Any, String, Nothing] =
+  ZPure.fail("fail")
+```
+
+Notice how now the error type is `String`, indicating that this computation can fail with a `String` error. In this case the success type is `Nothing` since we actually know that this computation can never succeed.
+
+If we are working with code that can throw exceptions we can use the `attempt` operator of `ZPure` to safely import that code into our computation, catching any non-fatal exceptions and translating them into failure values.
+
+```scala mdoc
+def parseIntThrowable(s: String): ZPure[Nothing, Unit, Unit, Any, Throwable, Int] =
+  ZPure.attempt(s.toInt)
+```
+
+We can use the `mapError` operator to transform the error type of a computation, for example to map it to a common error type we are using in other parts of our computation.
+
+```scala mdoc
+def parseInt(s: String): ZPure[Nothing, Unit, Unit, Any, String, Int] =
+  parseIntThrowable(s).mapError(_.getMessage)
+```
+
+We can recover from errors by using the `catchAll` operator, which allows us to recover from the failure using a new computation.
+
+```scala mdoc
+def parseIntOrDefault(s: String): ZPure[Nothing, Unit, Unit, Any, Nothing, Int] =
+  parseInt(s).catchAll(_ => ZPure.succeed(0))
+```
+
+Now the error type is `Nothing`, indicating that we have handled all our errors.
+
+There are many other error handling operators defined on `ZPure`. The `catchSome` operator allows us to only recover from certain errors.
+
+The `fold` and `foldM` operators allow us to handle both the failure and success cases at the same time. And the `orElse` operator allows us to specify a fallback computation that will be run if the original computation fails.
+
+One other concept from `ZIO` that carries over to `ZPure` is the ability to accumulate multiple errors.
+
+Normally when we use operators like `zip` or `flatMap` if an error occurs we will not go on to evaluate further parts of the computation until we get to an error handler that can potentially recover from it. This is typically what we want because if we have already failed there is no point in doing more work.
+
+However, `ZPure` also allows us to accumulate errors when we use the `zipWithPar` operator . This operator does not do actual parallelism, but is "parallel" in the sense that it will run both computations even if the first one fails and return any failures that occurred.
```

Comment:
Just a comment because it might be too late for any code change and I'm sure this question must have come up before.
If we want to avoid the possible confusion for having "Par" in `zipWithPar`, would `map2` be a better choice, which would also be familiar to people who've already used similar methods in other libraries?

Reply:
`map2` is the equivalent of `zipWith` which runs the left computation, then runs the right computation if that completes successfully, and combines the results. `zipWithPar` runs both computations regardless of whether one or both of them fails and then combines the results.